Build fails for 32bit linux/mips and linux/mipsle

**Is your feature request related to a problem? Please describe.**
Minio fails to compile for 32bit linux/mips and linux/mipsle architectures.
```bash
[minio]$ env GO111MODULE=on GOOS=linux GOARCH=mips go build -o minio-mips
# github.com/minio/minio/pkg/disk
pkg/disk/stat_linux.go:39:22: cannot use s.Type (type int32) as type int64 in argument to getFSType
# github.com/minio/minio/pkg/sys
pkg/sys/stats_linux.go:72:22: invalid operation: uint64(unit) * totalRAM (mismatched types uint64 and uint32)
```

**Describe the solution you'd like**
I know these architectures are not supported out of the box but it would be great if the code compiles effortless also for these 32bit architectures, especially since other 32bit arch already work.
A couple of substitutions in headers solve the problem for me: after these commands the above build step succeeds.
```bash
[minio]$ grep -rl ',!386' | xargs sed -i 's/,!386/,!386,!mips,!mipsle/g'
[minio]$ grep -rl 'linux,386' | xargs sed -i 's/linux,386/linux,386 linux,mips linux,mipsle/g'
```

It would be great to add this to the official repository, and maybe also take care of the other 32bit architectures that go offers a crosscompilation to.
Thank you for the great work by the way!

--- a/buildscripts/cross-compile.sh
+++ b/buildscripts/cross-compile.sh
@@ -9,7 +9,7 @@ function _init() {
     export CGO_ENABLED=0
 
     ## List of architectures and OS to test coss compilation.
-    SUPPORTED_OSARCH="linux/ppc64le linux/mips64 linux/arm64 linux/s390x darwin/arm64 darwin/amd64 freebsd/amd64 windows/amd64 linux/arm linux/386 netbsd/amd64"
+    SUPPORTED_OSARCH="linux/ppc64le linux/mips64 linux/arm64 linux/s390x darwin/arm64 darwin/amd64 freebsd/amd64 windows/amd64 linux/arm linux/386 netbsd/amd64 linux/mips"
 }
 
 function _build() {

--- a/pkg/disk/stat_linux.go
+++ b/pkg/disk/stat_linux.go
@@ -36,7 +36,7 @@ func GetInfo(path string) (info Info, err error) {
 		Free:   uint64(s.Frsize) * s.Bavail,
 		Files:  s.Files,
 		Ffree:  s.Ffree,
-		FSType: getFSType(s.Type),
+		FSType: getFSType(int64(s.Type)),
 	}
 	// Check for overflows.
 	// https://github.com/minio/minio/issues/8035

--- a/pkg/sys/stats_linux.go
+++ b/pkg/sys/stats_linux.go
@@ -69,7 +69,7 @@ func getSysinfoMemoryLimit() (limit uint64, err error) {
 
 	// Total RAM is always the multiplicative value
 	// of unit size and total ram.
-	return uint64(unit) * totalRAM, nil
+	return uint64(unit) * uint64(totalRAM), nil
 }
 
 // GetStats - return system statistics, currently only

